### PR TITLE
feat: Updated Keploy-config Documentation

### PR DESCRIPTION
**Pull request by Arindam200**
# Pull Request: 

## Description

This pull request enhances the documentation for Keploy-config to provide clearer explanations and examples for each configuration term. The goal is to improve user understanding and facilitate smoother usage of the Keploy-config YAML file.

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue).
- [x] Documentation update.
- [ ] New feature (non-breaking change which adds functionality).
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected).

## Additional Context

Screenshots/gifs: N/A

## Checklist:

- [ ] My code follows the style guidelines of this project.
- [ ] I have performed a self-review of my own code.
- [x] I have made corresponding changes to the documentation.
- [ ] I have added corresponding tests.
- [ ] Any dependent changes have been merged and published in downstream modules.

### CODE REVIEW

**Review by AkashKumar7902 — COMMENTED** [review]
please address the review comments

File: versioned_docs/version-2.0.0/running-keploy/configuration-file.md [hunk 1]
```diff
@@ -32,10 +36,18 @@ record:
...
+        filters:
+            - path: ""
+              urlMethods: []
+              headers: {}
+              host: ""
+    stubs:
+        filters:
+            - path: ""
+              host: "reqres.in"
+              port: 0
```
Comment (shivamsouravjha) [inline versioned_docs/version-2.0.0/running-keploy/configuration-file.md:50]: the one I gave was a sample config, you can remove the hose from here

File: versioned_docs/version-2.0.0/running-keploy/configuration-file.md [hunk 2]
```diff
@@ -44,65 +56,211 @@ test:
...
     withCoverage: false
     coverageReportPath: ""
```
Comment (shivamsouravjha) [inline versioned_docs/version-2.0.0/running-keploy/configuration-file.md:80]: same here , tge reqres.in can be removed and replace with empty string

File: versioned_docs/version-2.0.0/running-keploy/configuration-file.md [hunk 3]
```diff
@@ -44,65 +56,211 @@ test:
...
+    ```
+- **`stubs`**: Filters for recorded stubs.
+
+  Example:
+
+  ```yaml
+  stubs:
+    filters:
+      - path: ""
+        host: "reqres.in"
```
Comment (shivamsouravjha) [inline versioned_docs/version-2.0.0/running-keploy/configuration-file.md:141]: set host to be empty string

File: versioned_docs/version-2.0.0/running-keploy/configuration-file.md [hunk 4]
```diff
@@ -44,65 +56,211 @@ test:
...
+        urlMethods: []
+        headers: {}
+        host: ""
+    ```
+- **`stubs`**: Filters for recorded stubs.
+
+  Example:
+
+  ```yaml
+  stubs:
```
Comment (shivamsouravjha) [inline versioned_docs/version-2.0.0/running-keploy/configuration-file.md:138]: this is a bypass for mocks calls

File: versioned_docs/version-2.0.0/running-keploy/configuration-file.md [hunk 5]
```diff
@@ -44,65 +56,211 @@ test:
...
+- **`tests`**: Filters for tests.
+  Example:
+  ```
+  tests:
+  filters:
+    - path: ""
+      urlMethods: []
+      headers: {}
+      host: ""
+  ```
```
Comment (shivamsouravjha) [inline versioned_docs/version-2.0.0/running-keploy/configuration-file.md:189]: this is removed from tests , tests field not used in tests anymore

File: versioned_docs/version-2.0.0/running-keploy/configuration-file.md [hunk 6]
```diff
@@ -44,65 +56,211 @@ test:
...
+- **`stubs`**: Filters for stubs.
+
```
Comment (shivamsouravjha) [inline versioned_docs/version-2.0.0/running-keploy/configuration-file.md:191]: same as for the stubs during record

File: versioned_docs/version-2.0.0/running-keploy/configuration-file.md [hunk 7]
```diff
@@ -44,65 +56,211 @@ test:
...
+  ```
+  stubs:
+  filters:
+    - path: ""
+      host: "reqres.in"
+      port: 0
```
Comment (shivamsouravjha) [inline versioned_docs/version-2.0.0/running-keploy/configuration-file.md:197]: host to empty string

File: versioned_docs/version-2.0.0/running-keploy/configuration-file.md [hunk 8]
```diff
@@ -44,65 +56,211 @@ test:
...
+  ```
+
+- **`passThroughPorts`**: Ports passed through during testing.
+  Example: `passThroughPorts: [8080, 9000]`
+
```
Comment (shivamsouravjha) [inline versioned_docs/version-2.0.0/running-keploy/configuration-file.md:202]: removed this flag

File: versioned_docs/version-2.0.0/running-keploy/configuration-file.md [hunk 9]
```diff
@@ -44,65 +56,198 @@ test:
...
+- **`proxyport`**: Port number for the proxy. Default is 0.
+
+- **`containerName`**: Name of the container during recording.
+
+- **`networkName`**: Network name for the container during recording.
+
+- **`delay`**: Delay in seconds before recording each request. Default is 5 seconds.
+
+- **`passThroughPorts`**: Ports passed through during recording.
+
```
Comment (shivamsouravjha) [inline versioned_docs/version-2.0.0/running-keploy/configuration-file.md:120]: this field is removed now this works under stubs port.

File: versioned_docs/version-2.0.0/running-keploy/configuration-file.md [hunk 10]
```diff
@@ -44,65 +56,198 @@ test:
...
+- **`tests`**: Filters for recorded tests.
+
+  Example:
```
Comment (shivamsouravjha) [inline versioned_docs/version-2.0.0/running-keploy/configuration-file.md:123]: filters to record tests.

File: versioned_docs/version-2.0.0/running-keploy/configuration-file.md (hunk 9 above)
Comment (shivamsouravjha) [reply versioned_docs/version-2.0.0/running-keploy/configuration-file.md:120]: also you can explain some more about port, path and host in stubs and filters

Like if you see this 
```    # Example on using tests
    # tests: 
    #   filters:
    #    - path: "/user/app"
    #      urlMethods: ["GET"]
    #      headers: {
    #        "^asdf*": "^test"
    #      }
    #      host: "dc.services.visualstudio.com"
    # Example on using stubs
    # stubs: 
    #   filters:
    #    - path: "/user/app"
    #      port: 8080
    #    - port: 8081
    #    - host: "dc.services.visualstudio.com"
    #    - port: 8081
    #      host: "dc.services.visualstudio.com"
    #      path: "/user/app"
    #
    # Example on using globalNoise```
    
    user must be told how these fields can be used together as well as independently

File: versioned_docs/version-2.0.0/running-keploy/configuration-file.md [hunk 11]
```diff
@@ -44,65 +56,259 @@ test:
...
+  - The first stub applies to the path "/user/app" and the port 8080.
+  - The second stub applies to the port 8081.
+  - The third stub applies to the host "dc.services.visual
+## Advanced Noise Filtering:
+
+Earlier the only way to add the [noisy fields](http://keploy.io/docs/concepts/general-glossary/#3-noisy-field) was by modifying individual test file (testcase level). Now, With the introduction of config file, users can add the noisy fields at test-set and global level through config file itself.
+
+### Global Noise
+
+The Global Noise section is used to define parameters that are globally ignored for all API calls during testing. It enables you to filter out consistent noise, ensuring a cleaner evaluation of responses.
```
Comment (AkashKumar7902) [inline versioned_docs/version-2.0.0/running-keploy/configuration-file.md:265]: replace "The Global Noise section" to "The `global` subsection of `globalNoise`"

File: versioned_docs/version-2.0.0/running-keploy/configuration-file.md [hunk 12]
```diff
@@ -44,65 +56,259 @@ test:
...
+      "url": ["https?://\S+", "http://\S+"]
+    header: 
+      # To ignore the entire field, pass an empty array
+      "Date": []
+  test-sets:
 ```
 
-The `globalNoise` and `test-set noise` are optional fields in the config file. If not specified, the default value for both the fields is an empty object `{}`.
+1. **`global`**:
+  - **`body`**: Defines patterns to ignore for the request or response body, such as filtering out URLs. Example: `{"url": ["https?://\S+", "http://\S+"]}`
```
Comment (AkashKumar7902) [inline versioned_docs/version-2.0.0/running-keploy/configuration-file.md:280]: its only for response body, I believe

File: versioned_docs/version-2.0.0/running-keploy/configuration-file.md [hunk 13]
```diff
@@ -44,65 +56,259 @@ test:
...
+    # Ignore all the values for the "url" field
+    "uuid": ["b464d6df-d28b-4d12-8af1-0e2d61289578"]
+  header: 
+    # Pass the exact value to ignore for a field
+    "User-Agent": ["PostmanRuntime/7.34.0"]
+
+```
+
+**`test-set-1`**:
+  - **`body`**: Defines patterns to ignore for the request or response body within the specified test set. Example: `{"uuid": ["b464d6df-d28b-4d12-8af1-0e2d61289578"]}`
```
Comment (AkashKumar7902) [inline versioned_docs/version-2.0.0/running-keploy/configuration-file.md:301]: same here. only for response body

File: versioned_docs/version-2.0.0/running-keploy/configuration-file.md [hunk 14]
```diff
@@ -44,65 +56,259 @@ test:
...
+              host: ""
+              port: 0
     withCoverage: false
     coverageReportPath: ""
 ```
 
-The presence of this file allows the user to eliminate the monotonous way of specifying the same parameters/flags for each record or test command in every run. The flags specified in the Keploy's [CLI Command Docs](http://keploy.io/docs/running-keploy/cli-commands/) are now converted into parameters to the record and test command in the config file. These parameters take the same input as the flags in the CLI command. Specifying those parameters in the config file will allow the user to directly run the record or test command without specifying the flags as shown below:
+## Using the Config File
+
+The Keploy-config file eliminates the need to repeatedly specify parameters for each record or test command. The parameters in the file correspond to the flags in the Keploy [CLI Command Docs](http://keploy.io/docs/running-keploy/cli-commands/). To run a record or test command, use:
```
Comment (AkashKumar7902) [inline versioned_docs/version-2.0.0/running-keploy/configuration-file.md:85]: the last statement can be modified to :
"Using keploy-config can help reduce the record and test command to just:"

**Review by AkashKumar7902 — CHANGES_REQUESTED** [review]
please address the review comments

**Comment by AkashKumar7902** [comment]
LGTM

**Review by AkashKumar7902 — APPROVED** [review]

**Review by shivamsouravjha — APPROVED** [review]
LGTM

**Pull request merged**